### PR TITLE
[#175899260] API definition and integration test

### PR DESCRIPTION
**Pull request by infantesimone**
### Description

The goal of this PR is to add the following API definition (without business logic):

```yaml
"/payment-requests/{rptId}":
    x-swagger-router-controller: PagoPAProxyController
    parameters:
      - name: rptId
        in: path
        required: true
        description: Unique identifier for payments.
        type: string
    get:
      operationId: getPaymentInfo
      summary: Get Payment Info
      description: Retrieve information about a payment
      responses:
        "200":
          description: Payment information retrieved
          schema:
            "$ref": "#/definitions/PaymentRequestsGetResponse"
          examples:
            application/json:
              importoSingoloVersamento: 200,
              codiceContestoPagamento: "ABC123"
        "400":
          description: Bad request
          schema:
            $ref: "#/definitions/ProblemJson"
        "401":
          description: Bearer token null or expired.
        "500":
          description: PagoPA services are not available or request is rejected
          schema:
            $ref: "#/definitions/PaymentProblemJson"
  "/payment-activations":
    x-swagger-router-controller: PagoPAProxyController
    parameters:
      - name: test
        in: query
        description: Use test environment of PagoPAClient
        type: boolean
        required: false
    post:
      operationId: activatePayment
      summary: Activate Payment
      description: Require a lock (activation) for a payment
      parameters:
        - in: body
          name: body
          schema:
            $ref: "#/definitions/PaymentActivationsPostRequest"
          required: true
          x-examples:
            application/json:
              rptId: "12345678901012123456789012345"
              importoSingoloVersamento: 200
              codiceContestoPagamento: "ABC123"
      responses:
        "200":
          description: Payment activation process started
          schema:
            "$ref": "#/definitions/PaymentActivationsPostResponse"
          examples:
            application/json:
              importoSingoloVersamento: 200
        "400":
          description: Bad request
          schema:
            $ref: "#/definitions/ProblemJson"
        "401":
          description: Bearer token null or expired.
        "500":
          description: PagoPA services are not available or request is rejected
          schema:
            $ref: "#/definitions/PaymentProblemJson"
  "/payment-activations/{codiceContestoPagamento}":
    x-swagger-router-controller: PagoPAProxyController
    parameters:
      - name: codiceContestoPagamento
        in: path
        required: true
        description: Transaction Id used to identify the communication flow.
        type: string
      - name: test
        in: query
        description: Use test environment of PagoPAClient
        type: boolean
        required: false
    get:
      operationId: getActivationStatus
      summary: Get Activation status
      description: Check the activation status to retrieve the paymentId
      responses:
        '200':
          description: Payment information
          schema:
            $ref: "#/definitions/PaymentActivationsGetResponse"
          examples:
            application/json:
              idPagamento: "123455"
        "400":
          description: Invalid input
          schema:
            $ref: "#/definitions/ProblemJson"
        "401":
          description: Bearer token null or expired.
        "404":
          description: Activation status not found
          schema:
            $ref: "#/definitions/ProblemJson"
        "500":
          description: Unavailable service
          schema:
            $ref: "#/definitions/ProblemJson"
```

### Proposed changes

For more details see openapi defintion.

### CODE REVIEW

**Comment by pasqualespica** [comment]
LGTM 👍 just remember that `test` param has been removed now

**Comment by GiovanniMancini** [comment]
@infantesimone In my opinion api tests should be automated via `supertest` package or `docker` + `curl` (or similar pattern). What do you think about such testing strategy?

**Comment by infantesimone** [comment]
@GiovanniMancini good point, but is out of scope of PR. It will be useful to study `docker` + `curl` because _supertest_ is for _express apps_ but we have _express apps_  'nested' in _Azure Function App_, so i don't know if we can use it.

**Review by GiovanniMancini — APPROVED** [review]

**Review by cloudify — CHANGES_REQUESTED** [review]
Please add an openapi spec linter (i.e. `oval`) to te pipeline, see for example https://github.com/pagopa/io-backend/blob/master/.circleci/config.yml#L36

**Comment by GiovanniMancini** [comment]
> Please add an openapi spec linter (i.e. `oval`) to te pipeline, see for example https://github.com/pagopa/io-backend/blob/master/.circleci/config.yml#L36

@cloudify I have to apologize for my wrong idea of adding CI/CD pipelines in a separate branch when Azure Resources needed for deployment will have been up and running. Quite obviously code review pipelines should be added as soon as possible, so I added the one for backend function with a stage to do the `oval` check you requested

File: io-functions-pay-portal/.devops/code-review-be.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        steps:
+        - template: azure-templates/setup-project.yml
+        - script: yarn lint
+          workingDirectory: io-functions-pay-portal
+          displayName: 'Lint'
+      - job: validate_api_specification     
+        steps:
+        - script: npx oval validate -p io-functions-pay-portal/openapi/index.yaml
+          displayName: 'Validate API specification'
+      # - job: danger
```
Comment (cloudify) [inline io-functions-pay-portal/.devops/code-review-be.yml:57]: why is this commented out?

Comment (GiovanniMancini) [reply io-functions-pay-portal/.devops/code-review-be.yml:57]: I don't have the danger token, so I can't set up the danger control. I left it commented as a reminder

Comment (balanza) [reply io-functions-pay-portal/.devops/code-review-be.yml:57]: I've set the secret token on the pipeline configuration. 
Now you need to add the Danger job to the pipeline. You can just copypaste from https://github.com/pagopa/io-functions-template/blob/master/.devops/code-review-pipelines.yml#L53

Comment (GiovanniMancini) [reply io-functions-pay-portal/.devops/code-review-be.yml]: @balanza Please note that using the template you suggested - at least in that repo - results in the following weird error during the danger job:

`TypeError: global$1.XMLHttpRequest is not a constructor`

It seems to be related to a wrong `gitlab` version - an indirect dependency - which is fixed by upgrading danger to `danger@10.xx `. Please see the following reference: 

https://github.com/danger/danger-js/issues/884

**Comment by GiovanniMancini** [comment]
> Why this change ? what is the issue that led you to make this switch . ref [doc](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/bash?view=azure-devops)

Quite strangely the pipeline works on a linux image, but not on a windows image. During the build phase it stops complaining on syntax of  `generate` script


```
$ shx rm -rf dist && yarn generate
$ npm-run-all generate:*
$ shx rm -rf ./generated/definitions && mkdir -p ./generated/definitions && gen-api-models --api-spec ./openapi/index.yaml --out-dir ./generated/definitions
The syntax of the command is incorrect.
```

**Review by cloudify — APPROVED** [review]

**Pull request merged**